interfaces/builtin/wayland.go

```diff
@@ -56,6 +56,10 @@ owner /run/user/[0-9]*/wayland-[0-9]* rwk,
 # Allow access to common client Wayland sockets from non-snap clients
 /run/user/[0-9]*/{mesa,mutter,sdl,wayland-cursor,weston,xwayland}-shared-* rw,
 
+# Some Wayland based toolkits (Qt, GTK3, SDL2) create shm files to pass to the
+# server. Although they are passed by FD we still need rw access to the file.
+owner /run/user/[0-9]*/snap.*/{,wayland-*} rw,
```

Comment:
Is owner match correct here? Is there ever a time the passed fd will be opened by one user and accessed by another (eg, a non-root client and a root server).

Reply:
I don't envisage that being useful with snapped server & client. A daemon server and user client have unrelated $XDG_RUNTIME_DIRs, so they wouldn't connect.

It is, of course, possible to set up that configuration in the Classic environment.

Reply:
..although, it looks like the rule can be a lot more specific.

WIll update.